### PR TITLE
Extract message definitions in schema to allow referencing from other APIs

### PR DESCRIPTION
**Pull request by DiamondJoseph**

### CODE REVIEW

**Comment by DiamondJoseph** [comment]
Have made a pull request with the updated AsyncAPI parser to try and get the github action enabled, but will disable for now.

**Review by keithralphs — APPROVED** [review]
Looks good now

File: .github/workflows/api.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#name: API CI
```
Comment (callumforrester) [inline .github/workflows/api.yaml:1]: Would prefer this was deleted and put in an issue

**Review by callumforrester — CHANGES_REQUESTED** [review]
Would prefer CI job was deleted and put in an issue

**Review by callumforrester — APPROVED** [review]

**Pull request merged**